packages/styletron-engine-atomic/src/server/server.js

```diff
@@ -30,32 +31,44 @@ export type attrsT = {
 };
 
 export type sheetT = {|
+  insertBeforeMedia?: string,
   css: string,
   attrs: attrsT,
 |};
 
 type optionsT = {
   prefix?: string,
+  streamingMode?: boolean,
 };
 
 class StyletronServer implements StandardEngine {
   styleCache: MultiCache<{pseudo: string, block: string}>;
   keyframesCache: Cache<KeyframesObject>;
   fontFaceCache: Cache<FontFaceObject>;
-  styleRules: {[string]: string};
+  styleRules: {
+    [string]: {
+      insertBeforeMedia?: string,
+      rules: string,
+    },
+  };
   keyframesRules: string;
   fontFaceRules: string;
+  streamingMode: boolean;
 
   constructor(opts?: optionsT = {}) {
-    this.styleRules = {"": ""};
+    this.streamingMode = opts.streamingMode || false;
+    this.styleRules = {"": {rules: ""}};
     this.styleCache = new MultiCache(
       new SequentialIDGenerator(opts.prefix),
-      media => {
-        this.styleRules[media] = "";
+      (media, _cache, insertBeforeMedia) => {
+        this.styleRules[media] = {
+          ...(insertBeforeMedia && {insertBeforeMedia}),
```

Comment:
I'm thinking it should be OK if the value of this property is undefined

```suggestion
          insertBeforeMedia,
```